pkg/registry/common/setid/network_service_registry_server.go

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) 2020 Doc.ai and/or its affiliates.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package setid
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/golang/protobuf/ptypes/empty"
+	"github.com/google/uuid"
+	"github.com/networkservicemesh/api/pkg/api/registry"
+	"github.com/pkg/errors"
+
+	"github.com/networkservicemesh/sdk/pkg/registry/core/next"
+)
+
+// Some resource types require their names to follow the DNS label standard as defined in RFC 1123.
+// This means the name must contain at most 63 characters
+const maxNameLength = 63
+
+type networkServiceRegistryServer struct{}
+
+// NewNetworkServiceRegistryServer creates new instance of NetworkServiceRegistryServer which set the unique name for the endpoint on registration
+func NewNetworkServiceRegistryServer() registry.NetworkServiceRegistryServer {
+	return &networkServiceRegistryServer{}
+}
+
+func (n networkServiceRegistryServer) RegisterNSE(ctx context.Context, r *registry.NSERegistration) (*registry.NSERegistration, error) {
```

Comment:
Ah... so this is to set the id if the NSE doesn't?

Reply:
It fills NSE name if it did not set by previous chain elements. For example, if this element will be used with some k8s-implementation then this chain element will do nothing.
NOTE: In mono repo by default id sets by Kubernetes database https://github.com/networkservicemesh/networkservicemesh/blob/master/k8s/pkg/registryserver/nse.go#L86.

Reply:
:)